env.go

```diff
@@ -19,47 +62,32 @@ func (e *Env) Run(steward Steward) error {
 		return err
 	}
 
-	usageHint := ""
-	unsetVar := ""
-	setVar := ""
-	quoteReplacement := "\""
-	switch e.Shell {
-	case "fish":
-		usageHint = "# To load these variables into your shell, execute:\n#   eval (%s)\n"
-		unsetVar = "set -e %s;\n"
-		setVar = "set -x %s \"%s\";\n"
-		quoteReplacement = "\\\""
-	default:
-		usageHint = "# To load these variables into your shell, execute:\n#   eval $(%s)\n"
-		unsetVar = "unset %s\n"
-		setVar = "export %s=\"%s\"\n"
-		quoteReplacement = "\\\""
-	}
-
-	vars := env.Variables()
-
-	// sort the vars to unset
-	sort.Strings(vars.Unset)
+	templateStr := envFormatters["sh"]
 
-	// sort the vars to set
-	var keys []string
-	for key := range vars.Set {
-		keys = append(keys, key)
+	if e.Format == "shell" {
+		if format, ok := envFormatters[e.DetectedShell]; ok {
+			templateStr = format
+		}
+	} else {
+		if format, ok := envFormatters[e.Format]; ok {
+			templateStr = format
+		} else {
+			templateStr = e.Format
+		}
 	}
-	sort.Strings(keys)
 
-	// display the vars using the format string for the shell
-	if e.UsageHint {
-		fmt.Printf(usageHint, strings.Join(os.Args, " "))
-	}
+	variables := env.Variables()
+	vals := templateVals{}
 
-	for _, key := range vars.Unset {
-		fmt.Printf(unsetVar, key)
-	}
+	vals.Set = variables.Set
 
-	for _, key := range keys {
-		fmt.Printf(setVar, key, strings.Replace(vars.Set[key], "\"", quoteReplacement, -1))
-	}
+	unsets := variables.Unset
+	sort.Strings(unsets)
+	vals.Unset = unsets
+
+	vals.Command = e.Command
 
-	return nil
+	tmpl, err := template.New("envTmpl").Funcs(templateFuncMap).Parse(templateStr)
```

Comment:
I _might_ suggest hoisting this to where the format template is determined (before the creation of the `templateVals`).